### PR TITLE
Two files, CO2 comparison, tracking in multiple years

### PR DESCRIPTION
**Pull request by leeyap**
This PR addresses a few issues.

Structural change: there are now two .Rmd files, `trackingC_setup.Rmd` and `trackingC.Rmd`. The setup file creates the runlist, runs the model, and saves the output to csvs. These files are read in to `trackingC.Rmd`, where the visualization and analysis takes place (to do: update `readme` to clarify directory files and order of operations).

New things happening:
- Comparing Hector CO2 outputs to NOAA data. Currently, this produces two histograms for two time points, 1971 and 2021. This visualization is... striking, and Hector's CO2 output looks way below the NOAA numbers. May need a better graph here, could include CMIP6 CO2 data as well (to do?).
- Creating functions for the airborne fraction emissions calculations (to do: update comments so arguments and order are clear).
- Then, I've started to address #30. It seems like the outputs follow the same general trend, so the start date for tracking does not drastically impact model outcomes. 
- The next step is to look at the year or two following the tracking start date to calculate real AF in those years. I've created a simple point plot with those values, but am not sure where to take that info next. 

Feedback and suggestions welcome, thanks!

To-dos:

- [x] Update readme
- [x] Add CMIP CO2 data, fix graph
- [x] Clarify AF functions

### CODE REVIEW

**Comment by bpbond** [comment]
We can already see one big problem: the `trk_output.csv` file is 300 MB for only 100 runs! This implies it's going to be 3 GB for 1000 runs (at the limit of what we can read, probably) and 30 GB for 10,000 runs. 💥 

On my machine `read.csv` takes 40+ seconds to read this file:
```
> system.time(read.csv("./data_files/trk_output.csv"))
   user  system elapsed 
 43.013   0.411  43.723 
```

Ugh. Luckily we can do _much_ better by using either `read_csv` (from the readr package) or `fread` (from data.table):
```
> system.time(readr::read_csv("./data_files/trk_output.csv"))
user  system elapsed 
  6.383   0.320   1.758 

> system.time(data.table::fread("./data_files/trk_output.csv"))
   user  system elapsed 
  4.431   0.090   4.513 
```

More fundamentally, however, let's think about filtering the tracking output _before_ we write it out: I think there is tracking data know we're not going to ever need.

**Comment by bpbond** [comment]
The other issue is that the two Rmarkdown files share some common things (N_RUNS, the hector core, OUTPUT_YEARS) that we'd like to define just once, i.e. in a common file. Pushing a commit to address this and then we can talk next week!

**Comment by leeyap** [comment]
Re: filtering tracking data, I was going to select just `pool_name %in% c("earth_c", "atmos_c")`, but the destination fraction code requires all pools. For filtering by source name, we have a graph of composition of the atmosphere by source that also uses all source names. Not sure what other criteria to filter out by?

**Comment by bpbond** [comment]
>I was going to select just pool_name %in% c("earth_c", "atmos_c"), but the destination fraction code requires all pools. 

But only for earth_c, right?

For filtering by source name, we have a graph of composition of the atmosphere by source that also uses all source names.

Agreed, atmosphere needs the full suite of source information.

**Comment by leeyap** [comment]
Yes, only for `earth_c`, but I can't filter out by source since we need all the source info for the atmosphere graph - is that right? It's more than possible that I'm just not thinking of a better way

**Comment by bpbond** [comment]
@leeyap Re filtering I guess I was thinking we could do something like
```r
df %>%
  filter(source == "earth_c" ||  # we need all information about emissions
         pool == "atmos_c") # ...as well as atmosphere and its sources
```

**Comment by leeyap** [comment]
@bpbond  Ohh, I don't know why I was having such a hard time visualizing that. Done, and thanks! Would you mind taking a look over things before a merge?

File: trackingC_setup.Rmd [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+  message("  ")
+  # Set a tracking date, reset and run core
+  setvar(c, NA, TRACKING_DATE(), 1950, NA)
+  reset(c)
+  run(c)
+  # Access and save tracking data, model outputs
+  out <- list()
+  out$tdata <- get_tracking_data(c) %>% 
+    filter(year %in% OUTPUT_YEARS, 
+           source_name == "earth_c" | pool_name == "atmos_c")
```
Comment (bpbond) [inline trackingC_setup.Rmd:250]: Maybe a comment here, something like:
```r
# To keep file sizes down, we only save output that is either emissions (and all their destinations)
# or atmosphere (and all its sources)
```

**Comment by bpbond** [comment]
@leeyap Left you one additional suggestion but otherwise it looks 😍 — merge when ready!

**Comment by leeyap** [comment]
Thanks! Just a note, filtering added 0.12ish seconds per run to the runtime.

**Comment by bpbond** [comment]
> filtering added 0.12ish seconds per run

Oh, interesting. Fairly expensive but hopefully worth it; I wonder how big the resulting tracking file is now.

Nice work @leeyap 👏

**Pull request merged**